developer_manual/app/tutorial.rst

```diff
@@ -0,0 +1,487 @@
+========
+Tutorial
+========
+
+.. sectionauthor:: Bernhard Posselt <dev@bernhard-posselt.com>
+
+This tutorial will outline how to create a very simple notes app.
+
+
+Setup
+=====
+After the `development tool <https://github.com/owncloud/ocdev/blob/master/README.rst#installation>`_ has been installed the :doc:`development environment needs to be set up <../general/devenv>`. This can be done by either `downloading the zip from the website <http://owncloud.org/install/>`_ or cloning it directly from GitHub::
+
+    ocdev setup core --dir owncloud  --owncloud 8
+
+Now open another tab in the terminal and start the development server::
+
+    cd owncloud
+    php -S localhost:8080
+
+Afterwards the app can be created in the **apps** folder::
+
+    cd apps
+    ocdev startapp OwnNotes
+
+This creates a new folder called **ownnotes**. Now access and set up ownCloud through the webinterface at `http://localhost:8080 <http://localhost:8080>`_ and enable the OwnNotes application on the apps page.
+
+Routes & Controllers
+====================
+A typical web application consists of server side and client side code. The glue between those two parts are the URLs. In case of the notes app the following URLs will be used:
+
+* **GET /**: Returns the interface in HTML
+* **GET /notes**: Returns a list of all routes as in JSON
+* **GET /notes/1**: Returns a note with the id 1 in JSON
+* **DELETE /notes/1**: Deletes a note with the id 1
+* **POST /notes**: Creates a new note by passing in JSON
+* **PUT /notes/1**: Updates a note with the id 1 by passing in JSON
+
+On the client side we can call these URLs with the following jQuery code:
+
+.. code-block:: js
+
+    // example for calling the PUT /notes/1 URL
+    var baseUrl = OC.generateUrl('/apps/ownnotes');
+    var note = {
+        title: 'New note',
+        content: 'This is the note text'
+    };
+    var id = 1;
+    $.ajax({
+        url: baseUrl + '/notes/' + id,
+        type: 'PUT',
+        contentType: 'application/json',
+        data: JSON.stringify(note)
+    }).done(function (response) {
+        // handle success
+    }).fail(function (response, code) {
+        // handle failure
+    });
+
+On the server side we need to register a callback that is executed once the request comes in. The callback itself will be a method on a :doc:`controller <controllers>` and the controller will be connected to the URL with a :doc:`route <routes>`. The controller and route for the page are already set up in **ownnotes/appinfo/routes.php**:
+
+.. code-block:: php
+
+    <?php
+    return ['routes' => [
```

Comment:
Nothing changed regarding the old behavior. Only if an array is returned from the file, the new code will be executed.